pkg/ipam/crd.go

```diff
@@ -389,6 +437,12 @@ func (n *nodeStore) allocate(ip net.IP) (*ipamTypes.AllocationIP, error) {
 		return nil, fmt.Errorf("No IPs available")
 	}
 
+	if status, ok := n.ownNode.Status.IPAM.ReleaseIPs[ip.String()]; ok {
+		if status == ipamOption.IPAMMarkForRelease || status == ipamOption.IPAMReadyForRelease {
+			return nil, fmt.Errorf("IP not available, marked or ready for release")
+		}
+	}
```

Comment:
Sorry I didn't realize this last time, but a check like this also needs to be added to `nodeStore.allocateNext()`. Otherwise the IP will be handed out, as IPs are usually allocated via `allocateNext()`.

In addition, I think we also need to add a similar check to `hasMinimumIPsInPool`.

Reply:
Added checks to allocateNext() and updated `hasMinimumIPsInPool` to factor in IPs currently in handshake.